src/mlpack/core/data/detect_file_type.cpp

```diff
@@ -114,6 +119,42 @@ arma::file_type GuessFileType(std::istream& f)
     }
   }
 
+  if (hasComma && (hasBracket == false))
+  {
+    // If we believe we have a CSV file, then we want to try to skip any header
+    // row.  We'll detect a header row by simply seeing if anything in the first
+    // line doesn't parse as a number.
+    //
+    // TODO: this is not a foolproof algorithm, so there should eventually be a
+    // way added for the user to explicitly indicate that there is or isn't a
+    // header.
+    std::string firstLine;
+    std::getline(f, firstLine);
+
+    std::stringstream str(firstLine);
+    std::string token;
+    bool allNumeric = true;
+    // We'll abuse 'getline()' to split on commas.
+    while (std::getline(str, token, ','))
+    {
+      // Let's see if we can parse the token into a number.
+      try
+      {
+        (void) std::stod(token);
```

Comment:
I guess, this will only throw an exception if I pass `abc` but `1a` is fine. The solution I can think of is to use a stream instead:

```c++
std::stringstream s;
s << "-123.45f";
double num;
std::string rest;

s >> num >> rest;
if (rest.length() > 0)
  allNumeric = false;
```

Not sure that is necessary.

Reply:
Ah, excellent point.  I changed the strategy in bea27ff4.  (I had to adapt it a little bit; we have to check `s.fail()` after extracting `num`.)